lib/Sema/Constraint.cpp

```diff
@@ -504,6 +504,15 @@ DeclName Fix::getDeclNameArgument(ConstraintSystem &cs) const {
   return cs.FixedDeclNames[Data];
 }
 
+/// If this fix has optional result info, retrieve it.
+bool Fix::getAdditionallyOptionalArgument(ConstraintSystem &cs) const {
+  assert(getKind() == FixKind::UnwrapOptionalBase);
+
+  // Assumes that these fixes are always created in pairs, with the first
+  // created non-optional and the second with an added optional.
+  return (Data % 2) == 1;
```

Comment:
That's very clever. I feel like the name of this method is misleading, though. Perhaps `isUnwrapOptionalBaseByOptionalChaining()`? I know it's long, but that's what we're looking for---the chaining case.